table/block_based_table_reader.h

```diff
@@ -681,7 +675,7 @@ class BlockBasedTableIterator : public InternalIteratorBase<TValue> {
   void CheckOutOfBound();
 
  private:
-  BlockBasedTable* table_;
+  const BlockBasedTable* table_;
```

Comment:
I guess it's outside the scope of the PR but if it is const, why not make it const reference?

Reply:
Const ref would also work, assuming we don't mind the affected types becoming unassignable.